### PR TITLE
MINOR: Add some class javadoc to Admin client

### PR DESCRIPTION
**Pull request by tombentley**

### CODE REVIEW

**Comment by tombentley** [comment]
@chia7712 following your merge of #9397 perhaps you could review this? 

Very strictly speaking this is changing the javadoc of a public API, so it could be argued that it should have a KIP, but I don't think I've written anything at all contentious here, merely documenting the API that exists.

File: clients/src/main/java/org/apache/kafka/clients/admin/Admin.java [hunk 1]
```diff
@@ -45,15 +45,75 @@
...
+ * being unavailable.
+ * <p>
+ * Different operations necessitate requests being sent to different nodes in the cluster. For example
+ * {@link #createTopics(Collection)} communicates with the controller, but {@link #describeTopics(Collection)}
+ * can talk to any broker. When the recipient does not matter the instance will try to use the broker with the
+ * fewest outstanding requests.
+ * <p>
+ * The client will transparently retry certain errors which are usually transient.
+ * For example if the request for {@code createTopics()} got sent to a node which was not the controller
+ * the metadata would be refreshed and the request re-sent to the controller.
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/clients/admin/Admin.java:107]: request "got" re-sent to the control

File: clients/src/main/java/org/apache/kafka/clients/admin/Admin.java [hunk 2]
```diff
@@ -45,15 +45,75 @@
...
+ * <ul>
+ *     <li>Admin instances should be created using {@link Admin#create(Properties)} or {@link Admin#create(Map)}</li>
+ *     <li>Each operation typically has two overloaded methods, one which uses a default set of options and an
+ *     overloaded method where the last parameter is an explicit options object.
+ *     <li>The operation method's first parameter is a {@code Collection} of items to perform
+ *     the operation on. Batching multiple requests into a single call is more efficient and should be
+ *     preferred over multiple calls to the same method.
+ *     <li>The operation methods execute asynchronously.
+ *     <li>Each {@code xxx} operation method returns an {@code XxxResult} class with methods which expose
+ *     {@link org.apache.kafka.common.KafkaFuture KafkaFutures}
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/clients/admin/Admin.java]: {@link org.apache.kafka.common.KafkaFuture}

File: clients/src/main/java/org/apache/kafka/clients/admin/Admin.java [hunk 3]
```diff
@@ -45,15 +45,75 @@
...
+ *     for accessing the result(s) of the operation.
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/clients/admin/Admin.java]: this line can be added to tail of above line

File: clients/src/main/java/org/apache/kafka/clients/admin/Admin.java [hunk 4]
```diff
@@ -45,15 +45,75 @@
...
+ *     <li>Typically an {@code all()} method is provided for getting the overall success/failure of the batch and a
+ *     {@code values()} method provided access to each item in a request batch.
+ *     Other methods may also be provided.
+ *     <li>For synchronous behaviour use {@link org.apache.kafka.common.KafkaFuture#get}
+ * <p>
+ * Here is a simple example of using an Admin client instance to create a new topic:
+ * <pre>
+ * {@code
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/clients/admin/Admin.java:69]: lose a curly bracket?

**Review by chia7712 — COMMENTED** [review]
@tombentley I love this great comment. Some trivial comments are left.

File: clients/src/main/java/org/apache/kafka/clients/admin/Admin.java [hunk 5]
```diff
@@ -45,15 +45,76 @@
...
+ *     <li>The operation method's first parameter is a {@code Collection} of items to perform
+ *     the operation on. Batching multiple requests into a single call is more efficient and should be
+ *     preferred over multiple calls to the same method.
+ *     <li>The operation methods execute asynchronously.
+ *     <li>Each {@code xxx} operation method returns an {@code XxxResult} class with methods which expose
+ *     {@link org.apache.kafka.common.KafkaFuture} for accessing the result(s) of the operation.
+ *     <li>Typically an {@code all()} method is provided for getting the overall success/failure of the batch and a
+ *     {@code values()} method provided access to each item in a request batch.
+ *     Other methods may also be provided.
+ *     <li>For synchronous behaviour use {@link org.apache.kafka.common.KafkaFuture#get}
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/clients/admin/Admin.java]: ```{@link org.apache.kafka.common.KafkaFuture#get}``` => ```{@link org.apache.kafka.common.KafkaFuture#get()}```

**Review by chia7712 — COMMENTED** [review]
@tombentley one small comment left. overall, LGTM

**Comment by tombentley** [comment]
@chia7712 done. I also took the liberty of removing a couple of erroneous `@throws` tags.

File: clients/src/main/java/org/apache/kafka/clients/admin/Admin.java [hunk 6]
```diff
@@ -519,7 +519,6 @@ AlterConfigsResult incrementalAlterConfigs(Map<ConfigResource,
      *
      * @param replicaAssignment     The replicas with their log directory absolute path
      * @return                      The AlterReplicaLogDirsResult
-     * @throws InterruptedException Interrupted while joining I/O thread
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/clients/admin/Admin.java:461]: @dongjinleekr Could you please take a look at this ```throws```? It was added by your PR (https://github.com/apache/kafka/pull/7083).

Comment (dongjinleekr) [reply clients/src/main/java/org/apache/kafka/clients/admin/Admin.java:461]: @chia7712 Okay. Let me see. :smiley:

Comment (dongjinleekr) [reply clients/src/main/java/org/apache/kafka/clients/admin/Admin.java:461]: @chia7712 `InterruptedException` is not thrown anymore; This method does not call `KafkaFutureImpl#get` method anymore. It's safe to remove it.

**Review by chia7712 — APPROVED** [review]
@tombentley Thanks for this great patch. LGTM

@dongjinleekr Thanks for confirming!

**Pull request merged**